src/main/java/ca/fxco/pistonlib/blocks/pistons/basePiston/BasicPistonBaseBlock.java

```diff
@@ -240,13 +256,19 @@ public boolean triggerEvent(BlockState state, Level level, BlockPos pos, int typ
                     if (type != MotionType.PULL || frontState.isAir() ||
                             (frontState.getPistonPushReaction() != PushReaction.NORMAL && !frontState.is(ModTags.PISTONS)) ||
                             !canMoveBlock(frontState, level, frontPos, facing.getOpposite(), false, facing)) {
-                        level.removeBlock(headPos, false);
+                        if (!PistonLibConfig.illegalBreakingFix ||
+                                level.getBlockState(headPos).getDestroySpeed(level, headPos) != -1.0F) {
```

Comment:
Would it make sense to check if the block is the appropriate head block instead? This implementation does allow breaking blocks with headless pistons, only preventing breaking blocks you couldn't also break by hand.

Reply:
Well there's also a headless piston fix. This is for people who want to use headless pistons but not being able to break blocks that cannot usually be broken by hand with them.

Reply:
sounds good